app/src/main/java/com/concordium/wallet/ui/onramp/CcdOnrampSitesViewModel.kt

```diff
@@ -5,41 +5,93 @@ import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.viewModelScope
-import com.concordium.wallet.App
-import com.concordium.wallet.data.AccountRepository
+import com.concordium.wallet.data.backend.wert.WertRepository
+import com.concordium.wallet.ui.common.BackendErrorHandler
+import com.concordium.wallet.ui.onramp.swipelux.SwipeluxSettingsHelper
+import com.concordium.wallet.ui.onramp.wert.WertWidgetHelper
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 
 class CcdOnrampSitesViewModel(application: Application) : AndroidViewModel(application) {
     private val ccdOnrampSiteRepository: CcdOnrampSiteRepository by lazy(::CcdOnrampSiteRepository)
-    private val accountRepository: AccountRepository by lazy {
-        AccountRepository(App.appCore.session.walletStorage.database.accountDao())
-    }
+    private val wertRepository: WertRepository by lazy(::WertRepository)
 
     private val _listItemsLiveData = MutableLiveData<List<CcdOnrampListItem>>()
     val listItemsLiveData: LiveData<List<CcdOnrampListItem>> = _listItemsLiveData
 
-    var accountAddress: String? = null
+    private val _wertSite = MutableSharedFlow<Pair<CcdOnrampSite, Boolean>?>()
+    val wertSite = _wertSite.asSharedFlow()
+
+    private val _sessionLoading = MutableStateFlow(false)
+    val sessionLoading = _sessionLoading.asStateFlow()
+
+    private val _error = MutableSharedFlow<Int>(extraBufferCapacity = 1)
+    val error = _error.asSharedFlow()
+
+    lateinit var accountAddress: String
         private set
 
     init {
         postItems()
     }
 
-    fun initialize(accountAddress: String?) {
-        // Use the provided account address or, if none provided,
-        // the only one if there is one account.
-        if (accountAddress != null) {
-            this.accountAddress = accountAddress
-        } else {
-            viewModelScope.launch {
-                val allDoneAccounts = accountRepository.getAllDone()
-                if (allDoneAccounts.size == 1) {
-                    this@CcdOnrampSitesViewModel.accountAddress = allDoneAccounts.first().address
+    fun initialize(accountAddress: String) {
+        this.accountAddress = accountAddress
+    }
+
+    private fun getWertSessionId(accountAddress: String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            _sessionLoading.value = true
+            wertRepository.getWertSessionDetails(
+                walletAddress = accountAddress,
+                success = { response ->
+                    val wertSite = ccdOnrampSiteRepository.getSiteByName("Wert")
+                    viewModelScope.launch {
```

Comment:
If `getWertSessionDetails` were just a suspend function with a return type, you wouldn't have to nest callbacks and launches one into another